src/python/alias.cpp

```diff
@@ -116,8 +116,35 @@ static void set_variant(nb::args args) {
         );
     }
 
-    if (!curr_variant.equal(new_variant)) {
-        nb::object new_variant_module = variant_module(new_variant);
+    nb::object old_variant = curr_variant;
+    // For each requested _and_ available variant, in order of preference.
+    for (size_t i = 0; i < valid_variants.size(); ++i) {
+        const auto &requested_variant = valid_variants[i];
+        bool is_last = (i == valid_variants.size() - 1);
+
+        if (requested_variant.equal(old_variant)) {
+            // We're already using this variant, no need to do anything.
+            break;
+        }
+
+        nb::object new_variant_module;
+        try {
+            new_variant_module = variant_module(requested_variant);
+        } catch (const nb::python_error &e) {
+            // The variant failed to import, this could happen e.g. if the
+            // CUDA driver is installed, but there is no GPU available.
+            // We only allow such failures as long as we have more variants to try.
+            if (!is_last && e.matches(PyExc_ImportError)) {
+                // nb::str var_str = requested_variant;
+                // jit_log(LogLevel::Debug,
```

Comment:
I would have liked to make a debug log here, but `jit_log` is not linked against this module. Is there an easy way to log here?

Reply:
I see two options:
* We link `libmitsuba.so` to `mitsuba_alias.so` so that we can call `mitsuba::Thread::thread()->logger()->log(level, nullptr, __FILE__, __LINE__, ...)`
* We're guaranteed to have loaded `mitsuba_ext.so`, so we theoretically do a Python call to `mitsuba.Log(Level, message)` through Python.  

I'd personally prefer the second one.

Reply:
Thanks, doing a Log call through Python works :+1: